net/dnscache/dnscache.go

```diff
@@ -252,7 +250,26 @@ func (r *Resolver) lookupTimeoutForHost(host string) time.Duration {
 	return 10 * time.Second
 }
 
-func (r *Resolver) lookupIP(host string) (ip, ip6 net.IP, allIPs []net.IPAddr, err error) {
+func ip2addrs(n []net.IPAddr) []netip.Addr {
+	ips := make([]netip.Addr, 0, len(n))
+	for _, ip := range n {
+		if len(ip.IP) == 4 {
+			ips = append(ips, netip.AddrFrom4(*(*[4]byte)(ip.IP)))
+		} else {
+			ips = append(ips, netip.AddrFrom16(*(*[16]byte)(ip.IP)).WithZone(ip.Zone))
+		}
+	}
+	return ips
+}
+
+func ip2addr(ip net.IP) netip.Addr {
+	if len(ip) == 4 {
```

Comment:
netip.AddrFromSlice?

Reply:
or https://pkg.go.dev/go4.org/netipx?utm_source=godoc#FromStdIP

Reply:
There's a `bool` that's returned from that, which we could ignore, but then we'd have silent failures if something went wrong. With this we just fail loudly because something is very wrong if we're getting a `net.IP` that isn't either 4 or 16 bytes long.

Reply:
we could panic on `!ok` :)

Reply:
That's more lines of code to do the same thing :P

Reply:
But it makes readability better because I can go read docs in the library.

Reply:
I don't follow, how is reading the docs for `AddrFromSlice` any better than reading the docs for `AddrFrom4` and `AddrFrom16`? If we want it to have better readability I can add a doc comment to the function explaining what it does, but I don't think that changing this to use `AddrFromSlice` adds enough. As is this is pretty straightforward code.